Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1915,10 +1915,17 @@ def build_search_human_readable(args, parsed_search_results):
         if not isinstance(parsed_search_results[0], dict):
             headers = "results"
         else:
-            chosen_fields = []
-            for table_args in re.findall(r' table (?P<table>[^|]*)', args.get('query', '')):
-                chosen_fields.extend([field.strip('"')
-                                      for field in re.findall(r'((?:".*?")|(?:[^\s,]+))', table_args) if field])
+            query = args.get('query', '')
+            table_args = re.findall(r' {} (?P<{}>[^|]*)'.format('table', 'table'), query)
+            rename_args = re.findall(r' {} (?P<{}>[^|]*)'.format('rename', 'rename'), query)
+
+            chosen_fields = [field.strip('"') for arg_string in table_args
+                             for field in re.findall(r'((?:".*?")|(?:[^\s,]+))', arg_string) if field]
+            rename_dict = {field[0]: field[-1].strip('"') for arg_string in rename_args for field in
+                           re.findall(r'((?:".*?")|(?:[^\s,]+))( AS )((?:".*?")|(?:[^\s,]+))', arg_string) if field}
```

Comment:
Can we split it into two for loops? this is a bit unreadable.

Reply:
I've added  a comment to explain those lines because I think splitting them makes the code less clean and doesn't contribute to understanding the code. 

```suggestion
            chosen_fields = []
            for arg_string in table_args:
                 for field in re.findall(r'((?:".*?")|(?:[^\s,]+))', arg_string):
                     if field:
                         chosen_fields.append(field.strip('"'))
```